### ISSUE TITLE
Business clarification : Register workflow 

### ISSUE THREAD

**Issue by MarkAPhillips** [y0rk1e/Chrodie.WebApp#19]
When an user successfully registers - what is the defined workflow process?
1) Should a token be generated and returned so the user can be authorised to access the site - i.e. on successful registration should the user be redirected to the dashboard - if so a token should be returned.
2) If not should the user register and then have to login again - if this is the case a verification link should be sent to the users email address which they click on to be able to confirm who they are are then login.

**Comment by MarkAPhillips** [comment]
Noticed I received this email:

Confirm your account‏

To: work-mpconsultants@outlook.com
DoNotReply@Chrodie.co.uk
Please confirm your account by clicking here

**Comment by y0rk1e** [comment]
Yes currently, if you don't validate your email you will get an error message at logon stating your email address hasn't been validated.

**Comment by MarkAPhillips** [comment]
OK - we need to be able to define a link dependent on environment that a user can go to 

e.g. /register-validate/{encrypted id}

**Comment by y0rk1e** [comment]
Is that not already in the link in the email. It creates a URL which encodes the user id and the email validation code. The API then validates the code against the user id and if they are correct marks the user EmailValidated = true.

Is this the functionality your asking for? Or have I missed something?

**Comment by MarkAPhillips** [comment]
When the user clicks on that link - all you get is a blank page so from a UX perspective an user is unsure what has happened.
It would be better that the link takes you to a page on the website that then calls the Web API endpoint that on success displays a message to the user - stating their account is now validated.

If the link has expired or has already been validated - a standard message would be displayed to the  user stating the link is invalid etc.

**Comment by y0rk1e** [comment]
Ok I understand.

You create the UI URL that I use in the link in the email (with the user id and code). Navigating to that URL, the UI code pulls out the parameters and calls the WebAPI endpoint that is currently in the email. The UI receives the response and displays success failure messages to the user.
